crates/notion-core/src/tool.rs

```diff
@@ -330,4 +334,17 @@ impl Tool for Yarn {
     fn command(self) -> Command {
         self.0
     }
+
+    /// Perform any tasks which must be run after the tool runs but before exiting.
+    fn finalize(session: &Session, maybe_status: &io::Result<ExitStatus>) {
+        if let Ok(_) = maybe_status {
+            if let Some(project) = session.project() {
+                let errors = project.autoshim();
+
+                for error in errors {
+                    display_error(&error);
```

Comment:
Since autoshimming is running essentially behind the scenes, and these errors are not affecting the status, should these be displayed as warnings instead of errors?

Reply:
I disagree.  Auto-shimming is an important part of Notion's core functionality and I think it needs to be called out as a significant problem when it fails.  Doubly so *because* it won't affect the tool's exit status.

Reply:
This was resolved in last week's call.